### PR TITLE
SQL homework 

### PR DESCRIPTION
**Pull request by Valelaaa**
-- first 18 questions

### CODE REVIEW

File: homework-selects/sql_questions.sql [hunk 1]
```diff
@@ -3,55 +3,523 @@
...
+       COUNT(*)
+FROM employees
+    INNER JOIN departments
+        USING(department_id)
+    WHERE commission_pct
+        IS NOT NULL
+GROUP BY department_name;
+
 --12. job title and average salary of employees.
+SELECT job_title AS job,
```
Comment (NataliaIovva) [inline homework-selects/sql_questions.sql:97]: When you count average salary of employees you should consider each employee, but you get an average salary which employee can get at a job

File: homework-selects/sql_questions.sql [hunk 2]
```diff
@@ -3,55 +3,523 @@
...
+WHERE salary >
+      (SELECT AVG(salary)
+       FROM employees);
+
 --17. the employee name ( first name and last name ), employee id and salary of all employees who report to Payam.
+SELECT first_name || ' ' || last_name AS name,
+       employee_id, salary
+FROM employees
+WHERE manager_id =
+      (SELECT manager_id
```
Comment (NataliaIovva) [inline homework-selects/sql_questions.sql:144]: here should be employee id of Payam

File: homework-selects/sql_questions.sql [hunk 3]
```diff
@@ -3,55 +3,523 @@
...
+SELECT *
+FROM employees
+         WHERE salary =
+                (SELECT MAX(salary)
+                 FROM employees
+                WHERE salary <
+                      (SELECT MAX(salary)
+                       FROM employees))
+                        ORDER BY DBMS_RANDOM.RANDOM
+                        FETCH FIRST ROWS ONLY;
```
Comment (NataliaIovva) [inline homework-selects/sql_questions.sql:196]: here you don't have condition to get just one employee's information

File: homework-selects/sql_questions.sql [hunk 4]
```diff
@@ -3,55 +3,523 @@
...
+--23. the employee name( first name and last name ) and hire
+-- date for all employees in the same department as Clara. Exclude Clara.
+SELECT first_name || ' ' || last_name AS name,
+       hire_date AS hireDate
+FROM employees
+WHERE department_id IN
+                     (SELECT department_id
+                      FROM employees
+                      WHERE first_name LIKE 'Clara')
+                 AND first_name NOT LIKE 'Clara';
```
Comment (NataliaIovva) [inline homework-selects/sql_questions.sql:207]: after LIKE usually goes regex if you have exactly value use =

File: homework-selects/sql_questions.sql [hunk 5]
```diff
@@ -3,55 +3,523 @@
...
+
 --24. the employee number and name( first name and last name ) for all employees who work in a department with any employee whose name contains a T.
+SELECT employee_id AS employeeNumber,
+       first_name || ' ' || last_name AS name
+FROM employees
+WHERE department_id IN
+      (SELECT department_id
+       FROM employees
+       WHERE UPPER(first_name || last_name) LIKE '%T%'
```
Comment (NataliaIovva) [inline homework-selects/sql_questions.sql:216]: UPPER(first_name || last_name) LIKE '%T%'
here you will get true in brackets true and compare the result of UPPER(first_name || last_name) with regex '%T%' so you get employees not only which cintains T in first or last name

File: homework-selects/sql_questions.sql [hunk 6]
```diff
@@ -3,55 +3,523 @@
...
 --25. full name(first and last name), job title, starting and ending date of last jobs for those employees with worked without a commission percentage.
---26. the employee number, name( first name and last name ), and salary for all employees who earn more than the average salary and who work in a department with any employee with a J in their name.
---27. the employee number, name( first name and last name ) and job title for all employees whose salary is smaller than any salary of those employees whose job title is MK_MAN.
---28. the employee number, name( first name and last name ) and job title for all employees whose salary is smaller than any salary of those employees whose job title is MK_MAN. Exclude Job title MK_MAN.
+SELECT first_name || ' ' || last_name AS name,
+       job_title AS jobTitle,
+       start_date, end_date
+FROM job_history
+    LEFT JOIN jobs
+        USING(job_id)
```
Comment (NataliaIovva) [inline homework-selects/sql_questions.sql:225]: missing condition with start date and end date

File: homework-selects/sql_questions.sql [hunk 7]
```diff
@@ -3,55 +3,523 @@
...
+       first_name || ' ' || last_name AS name,
+       salary
+FROM employees
+WHERE salary >
+      (SELECT AVG(salary)
+       FROM employees)
+  AND department_id IN
+      (SELECT department_id
+       FROM employees
+       WHERE UPPER(first_name || ' ' || last_name) LIKE '%J%');
```
Comment (NataliaIovva) [inline homework-selects/sql_questions.sql:243]: rewrite condition `UPPER(first_name || ' ' || last_name) LIKE '%J%'`

File: homework-selects/sql_questions.sql [hunk 8]
```diff
@@ -3,55 +3,523 @@
...
+       job_title
+FROM employees
+    LEFT JOIN jobs
+        USING (job_id)
+WHERE salary <
+      (SELECT salary
+       FROM employees
+           LEFT JOIN jobs
+               USING(job_id)
+       WHERE job_id LIKE 'MK_MAN');
```
Comment (NataliaIovva) [inline homework-selects/sql_questions.sql:258]: here use '=' instead LIKE

File: homework-selects/sql_questions.sql [hunk 9]
```diff
@@ -3,55 +3,523 @@
...
+SELECT phone_number,
+       first_name || ' ' || last_name AS name,
+       job_title
+FROM employees
+    LEFT JOIN jobs
+        USING (job_id)
+WHERE salary <
+      (SELECT salary
+       FROM employees
+           LEFT JOIN jobs
```
Comment (NataliaIovva) [inline homework-selects/sql_questions.sql:272]: you already have JOB_ID in Employee table so I think JOIN with jobs is redundant

File: homework-selects/sql_questions.sql [hunk 10]
```diff
@@ -3,55 +3,523 @@
...
+-- HIGH and LOW respectively for those employees whose salary
+-- is more than and less than the average salary of all employees.
+SELECT employee_id,
+       first_name || ' ' || last_name AS name,
+       salary,
+       CASE WHEN salary >
+             (SELECT AVG(salary)
+              FROM employees)
+            THEN 'HIGH'
+        ELSE 'LOW'
```
Comment (NataliaIovva) [inline homework-selects/sql_questions.sql:320]: if you use just 'else' LOW you will get also salary which is equal to avg, but in condition is written only abbout more than and less then

File: homework-selects/sql_questions.sql [hunk 11]
```diff
@@ -3,55 +3,523 @@
...
+       WHERE last_name LIKE 'Ozer%');
+
 --36. which employees have a manager who works for a department based in the US.
+SELECT first_name || ' ' || last_name AS name
+FROM employees
+    LEFT JOIN departments
+        USING(manager_id)
+LEFT JOIN locations
+    USING(location_id)
+LEFT JOIN countries
```
Comment (NataliaIovva) [inline homework-selects/sql_questions.sql:367]: you can use
'WHERE l.country_id = 'US';'
and do not make last join

File: homework-selects/sql_questions.sql [hunk 12]
```diff
@@ -3,55 +3,523 @@
...
---45. the first and last name, salary, and department ID for those employees who earn less than the minimum salary of a department which ID is 70.
---46. the first and last name, salary, and department ID for those employees who earn less than the average salary, and also work at the department where the employee Laura is working as a first name holder.
+--the employees who gets the salary as the salary earn by the employee
+-- which is maximum within the joining person January 1st, 2002 and December 31st, 2003.
+SELECT a.employee_id,
+       a.first_name || ' ' || a.last_name AS name,
+       a.salary,
+       department_name,
+       city
+FROM locations
```
Comment (NataliaIovva) [inline homework-selects/sql_questions.sql:387]: when you have to collect more information from employee table, better practice will be when first 'WHERE' is from employee and then from that all information you collect just needed

File: homework-selects/sql_questions.sql [hunk 13]
```diff
@@ -3,55 +3,523 @@
...
+--41. the department name and Id for all departments where they located,
+-- that Id is equal to the Id for the location where department number 30 is located.
+SELECT department_name, department_id
+       FROM departments
+LEFT JOIN locations
+USING (location_id)
+WHERE location_id =
+      (SELECT location_id
+       FROM departments
+       LEFT JOIN locations
```
Comment (NataliaIovva) [inline homework-selects/sql_questions.sql:428]: that left join is redundant

File: homework-selects/sql_questions.sql [hunk 14]
```diff
@@ -3,55 +3,523 @@
...
+WHERE department_id =
+      (SELECT department_id
+       FROM employees
+       WHERE employee_id = 201);
+
+--43. the first and last name, salary, and department ID for those employees
+-- whose salary is equal to the salary of the employee who works in that department which ID is 40.
+SELECT first_name,last_name,department_id
+FROM employees
+WHERE salary =
```
Comment (NataliaIovva) [inline homework-selects/sql_questions.sql:446]: here should be in instead of =

File: homework-selects/sql_questions.sql [hunk 15]
```diff
@@ -3,55 +3,523 @@
...
+       WHERE department_id = 40);
+
+--45. the first and last name, salary, and department ID for those employees
+-- who earn less than the minimum salary of a department which ID is 70.
+SELECT first_name, last_name, salary,department_id
+FROM employees
+WHERE salary <
+      (SELECT min_salary
+       FROM employees
+                LEFT JOIN jobs
```
Comment (NataliaIovva) [inline homework-selects/sql_questions.sql:469]: that join is redundant

File: homework-selects/sql_questions.sql [hunk 16]
```diff
@@ -3,55 +3,521 @@
...
+       department_name AS department, city,
+       state_province AS stateProvince
+FROM employees
+         LEFT JOIN departments USING (department_id)
+         LEFT JOIN locations USING (location_id);
+
 -- 3. the first name, last name, department number and department name, for all employees for departments 80 or 40.
+SELECT first_name AS firstName,
+       last_name AS lastName,
+       department_id AS departmentNumber,
```
[the remark this answers is not in the document]
Comment (Valelaaa) [reply homework-selects/sql_questions.sql:19]: Understood!

File: homework-selects/sql_questions.sql [hunk 17]
```diff
@@ -3,55 +3,521 @@
...
+       department_name AS department
+FROM employees
+         LEFT JOIN departments
+             USING (department_id)
+WHERE department_id = 80
+   OR department_id = 40;
```
[the remark this answers is not in the document]
Comment (Valelaaa) [reply homework-selects/sql_questions.sql:25]: fixed

File: homework-selects/sql_questions.sql [hunk 18]
```diff
@@ -3,55 +3,521 @@
...
+SELECT first_name, last_name, email,
+       phone_number, department_name
+FROM employees
+    INNER JOIN departments
+        USING(manager_id);
+
+-- 9. the first name, last name, and department number for those
+-- employees who works in the same department as the employee who
+-- holds the last name as Taylor.
+SELECT first_name, last_name, department_id
```
[the remark this answers is not in the document]
Comment (Valelaaa) [reply homework-selects/sql_questions.sql:71]: added

File: homework-selects/sql_questions.sql [hunk 19]
```diff
@@ -3,55 +3,521 @@
...
+FROM employees
+WHERE department_id IN
+      (SELECT department_id
+       FROM employees
+       WHERE last_name LIKE 'Taylor');
```
[the remark this answers is not in the document]
Comment (Valelaaa) [reply homework-selects/sql_questions.sql:76]: changed

File: homework-selects/sql_questions.sql [hunk 20]
```diff
@@ -3,55 +3,521 @@
...
+FROM employees
+    INNER JOIN departments
+        USING(department_id)
+    WHERE commission_pct
+        IS NOT NULL
+GROUP BY department_name;
+
 --12. job title and average salary of employees.
+SELECT job_title AS job,
+       (max_salary+min_salary)/2 AS avgSalary
```
[the remark this answers is not in the document]
Comment (Valelaaa) [reply homework-selects/sql_questions.sql]: Fixed

File: homework-selects/sql_questions.sql [hunk 21]
```diff
@@ -3,55 +3,523 @@
...
 -- 7. the first name of all employees and the first name of their manager including those who does not working under any manager.
+SELECT a.first_name AS employee,
+       b.first_name AS manager
+FROM employees a
+         LEFT JOIN employees b
+                   ON a.manager_id = b.employee_id;
+
 -- 8. the details of employees who manage a department.
--- 9. the first name, last name, and department number for those employees who works in the same department as the employee who holds the last name as Taylor.
+SELECT first_name, last_name, email,
```
[the remark this answers is not in the document]
Comment (Valelaaa) [reply homework-selects/sql_questions.sql:62]: Got it

File: homework-selects/sql_questions.sql [hunk 22]
```diff
@@ -3,55 +3,523 @@
...
+SELECT first_name, last_name, department_id
+FROM employees
+WHERE department_id IN
+      (SELECT department_id
+       FROM employees
+       WHERE last_name LIKE 'Taylor');
+
 --10. the department name and number of employees in each of the department.
---11. the name of the department, average salary and number of employees working in that department who got commission.
+SELECT department_name, COUNT(*)
```
[the remark this answers is not in the document]
Comment (Valelaaa) [reply homework-selects/sql_questions.sql:79]: Done

File: homework-selects/sql_questions.sql (hunk 1 above)
Comment (Valelaaa) [reply homework-selects/sql_questions.sql:97]: Got it

File: homework-selects/sql_questions.sql (hunk 2 above)
Comment (Valelaaa) [reply homework-selects/sql_questions.sql:144]: Fixed

File: homework-selects/sql_questions.sql (hunk 3 above)
Comment (Valelaaa) [reply homework-selects/sql_questions.sql:196]: Got it

File: homework-selects/sql_questions.sql (hunk 4 above)
Comment (Valelaaa) [reply homework-selects/sql_questions.sql:207]: Understood

File: homework-selects/sql_questions.sql (hunk 5 above)
Comment (Valelaaa) [reply homework-selects/sql_questions.sql:216]: Done

File: homework-selects/sql_questions.sql (hunk 6 above)
Comment (Valelaaa) [reply homework-selects/sql_questions.sql:225]: done

File: homework-selects/sql_questions.sql (hunk 7 above)
Comment (Valelaaa) [reply homework-selects/sql_questions.sql:243]: fixed

File: homework-selects/sql_questions.sql (hunk 8 above)
Comment (Valelaaa) [reply homework-selects/sql_questions.sql:258]: Got it

File: homework-selects/sql_questions.sql (hunk 9 above)
Comment (Valelaaa) [reply homework-selects/sql_questions.sql:272]: Fixed

File: homework-selects/sql_questions.sql (hunk 10 above)
Comment (Valelaaa) [reply homework-selects/sql_questions.sql:320]: Got It

File: homework-selects/sql_questions.sql (hunk 11 above)
Comment (Valelaaa) [reply homework-selects/sql_questions.sql:367]: Done

File: homework-selects/sql_questions.sql (hunk 13 above)
Comment (Valelaaa) [reply homework-selects/sql_questions.sql:428]: Got it

File: homework-selects/sql_questions.sql (hunk 15 above)
Comment (Valelaaa) [reply homework-selects/sql_questions.sql:469]: Got It

File: homework-selects/sql_questions.sql (hunk 14 above)
Comment (Valelaaa) [reply homework-selects/sql_questions.sql:446]: Fixed

File: homework-selects/sql_questions.sql (hunk 6 above)
Comment (NataliaIovva) [reply homework-selects/sql_questions.sql:222]: try to start in this select form employee table

File: homework-selects/sql_questions.sql [hunk 23]
```diff
@@ -3,55 +3,522 @@
...
+                 AND first_name != 'Clara';
+
 --24. the employee number and name( first name and last name ) for all employees who work in a department with any employee whose name contains a T.
+SELECT employee_id AS "employee number",
+       first_name || ' ' || last_name AS name
+FROM employees
+WHERE department_id IN
+      (SELECT department_id
+       FROM employees
+       WHERE first_name LIKE '%T%');
```
[the remark this answers is not in the document]
Comment (Valelaaa) [reply homework-selects/sql_questions.sql]: Ok

File: homework-selects/sql_questions.sql (hunk 12 above)
Comment (Valelaaa) [reply homework-selects/sql_questions.sql:390]: Done

**Review by NataliaIovva — APPROVED** [review]

**Pull request merged**